src/supremm/datasource/factory.py

```diff
@@ -0,0 +1,22 @@
+from supremm.datasource.pcp.pcpdatasource import PCPDatasource
+from supremm.datasource.prometheus.promdatasource import PromDatasource
+
+
+class DatasourceFactory():
+    """ Datasource class helper """
+
+    def __init__(self, preprocs, plugins, resconf):
+
+        if resconf["datasource"] == "pcp":
+            self._datasource = PCPDatasource(preprocs, plugins)
+        elif resconf["datasource"] == "prometheus":
+            self._datasource = PromDatasource(preprocs, plugins, resconf)
```

Comment:
Perhaps throw an exception with a useful error message if the resconf['datasource'] is not a valid value.

Reply:
Resolved in https://github.com/ubccr/supremm/pull/259/commits/bcea3a659671902a42e250936536b72d96e6a352, https://github.com/ubccr/supremm/pull/259/commits/cfcd5ba2f7f6785e91664f0325fa17493056f74f, and https://github.com/ubccr/supremm/pull/259/commits/fc08e7a5784945738b8c4973213d153511b7c787